Indent should ignore indentation of variable assigment in multiline variable declaration

**Tell us about your environment**
* **ESLint Version:**
master
* **Node Version:**
6.10
* **npm Version:**
3.10


**What parser (default, Babel-ESLint, etc.) are you using?**
typescript
**Please show your full configuration:**

```js
module.exports = {
//    "extends": ["eslint:all"],
    "parser": "typescript-eslint-parser",
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "rules": {
        "indent": "error"
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

Invalid
```js
const fooLongName
        = 1,
    barLongName
        = 2,
    bazLongName
        = 3;
```

**What did you expect to happen?**

I expected this to be a valid indentation style. The indent-legacy rule allows for this style. Also if equal operators are on the previous line it is considered a valid indentation.

Valid
```js
const fooLongName =
        1,
    barLongName =
        2,
    bazLongName =
        3;
```



**What actually happened? Please include the actual, raw output from ESLint.**
It is expected to be indented like so:

```js
const fooLongName
    = 1,
    barLongName
    = 2,
    bazLongName
    = 3;
```

```
/mnt/c/Users/reyad.attiyat/Workspace/eslint-test/test-var-indent.ts
  2:1  error  Expected indentation of 4 spaces but found 8  indent
  4:1  error  Expected indentation of 4 spaces but found 8  indent
  6:1  error  Expected indentation of 4 spaces but found 8  indent
```

I can a create a PR to fix this. Just want to confirm this is not intended behavior. When the equal operator is on the line with the variable identifier the assignment indentation is ignored completely. We should probably do the same when the equal operator is on the same line as the assignment.

I have tested the changes made in https://github.com/eslint/eslint/pull/8498 and it does not fix this case.

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -1039,10 +1039,10 @@ module.exports = {
 
             VariableDeclarator(node) {
                 if (node.init) {
-                    const operator = sourceCode.getTokenBefore(node.init);
+                    const equalOperator = sourceCode.getTokenBefore(node.init, astUtils.isNotOpeningParenToken);
 
-                    if (operator.type === "Punctuator" && operator.value === "=" && tokenInfo.isFirstTokenOfLine(operator)) {
-                        offsets.ignoreToken(operator);
+                    if (equalOperator) {
+                        offsets.ignoreToken(equalOperator);
                     }
                     offsets.ignoreToken(sourceCode.getFirstToken(node.init));
                 }

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -1039,6 +1039,11 @@ module.exports = {
 
             VariableDeclarator(node) {
                 if (node.init) {
+                    const operator = sourceCode.getTokenBefore(node.init);
+
+                    if (operator.type === "Punctuator" && operator.value === "=" && tokenInfo.isFirstTokenOfLine(operator)) {
+                        offsets.ignoreToken(operator);
+                    }
                     offsets.ignoreToken(sourceCode.getFirstToken(node.init));
                 }
             },

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -1041,9 +1041,7 @@ module.exports = {
                 if (node.init) {
                     const equalOperator = sourceCode.getTokenBefore(node.init, astUtils.isNotOpeningParenToken);
 
-                    if (equalOperator) {
-                        offsets.ignoreToken(equalOperator);
-                    }
+                    offsets.ignoreToken(equalOperator);
                     offsets.ignoreToken(sourceCode.getFirstToken(node.init));
                 }
             },

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -559,6 +559,15 @@ ruleTester.run("indent", rule, {
         },
         {
             code: unIndent`
+                var foo
+                  = (1),
+                  bar
+                    = (2)
+            `,
+            options: [2, { VariableDeclarator: 1 }]
+        },
+        {
+            code: unIndent`
                 var foo = 1,
                     bar = 2,
                     baz = 3

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -532,6 +532,34 @@ ruleTester.run("indent", rule, {
         {
             code: unIndent`
                 var foo = 1,
+                  bar
+                    = 2
+            `,
+            options: [2, { VariableDeclarator: 1 }]
+        },
+        {
+            code: unIndent`
+                var foo
+                  = 1,
+                  bar
+                    = 2
+            `,
+            options: [2, { VariableDeclarator: 1 }]
+        },
+        {
+            code: unIndent`
+                var foo
+                  =
+                  1,
+                  bar
+                    =
+                    2
+            `,
+            options: [2, { VariableDeclarator: 1 }]
+        },
+        {
+            code: unIndent`
+                var foo = 1,
                     bar = 2,
                     baz = 3
                 ;

